### PR TITLE
Fix `stoi` out_of_range issue

### PR DESCRIPTION
**Pull request by vurusovs**
### Tickets:
 - CVS-118406

### CODE REVIEW

**Review by ilyachur — APPROVED** [review]

File: src/tests/test_utils/common_test_utils/src/common_utils.cpp [hunk 1]
```diff
@@ -70,32 +72,34 @@ size_t getVmRSSInKB() {
...
-        if (isdigit(c))
-            res += c;
-    if (res.empty())
-        // If number wasn't found return -1
-        return -1;
-    return std::stoi(res);
-}
-
 size_t getSystemDataByName(char* name) {
+    auto parseLine = [](std::string line) -> size_t {
```
Comment (ilyachur) [inline src/tests/test_utils/common_test_utils/src/common_utils.cpp:76]: Why do you need this code?
Maybe could you use `<<` operator from stream to get size_t?

**Review by ilyachur — CHANGES_REQUESTED** [review]

**Review by ilyachur — APPROVED** [review]

File: src/tests/test_utils/common_test_utils/src/common_utils.cpp [hunk 2]
```diff
@@ -56,7 +56,9 @@ std::string generateTestFilePrefix() {
 static PROCESS_MEMORY_COUNTERS getMemoryInfo() {
     static PROCESS_MEMORY_COUNTERS pmc;
     pmc.cb = sizeof(PROCESS_MEMORY_COUNTERS);
-    GetProcessMemoryInfo(GetCurrentProcess(), &pmc, pmc.cb);
+    auto status = GetProcessMemoryInfo(GetCurrentProcess(), &pmc, pmc.cb);
+    if (!status)
```
[the remark this answers is not in the document]
Comment (vurusovs) [reply src/tests/test_utils/common_test_utils/src/common_utils.cpp:60]: Done

File: src/tests/test_utils/common_test_utils/src/test_common.cpp [hunk 3]
```diff
@@ -30,10 +30,12 @@ TestsCommon::TestsCommon()
     : PGLink(new utils::PostgreSQLLink(this))
 #endif
 {
+#ifndef __APPLE__  // TODO: add getVmSizeInKB() for Apple platform
```
Comment (vurusovs) [inline src/tests/test_utils/common_test_utils/src/test_common.cpp:33]: @ilyachur I decided to remove use `getVmSizeInKB()` from here in order to throw clear exception in any other use (some other test which will rely on it's implementation)

**Review by ilyachur — APPROVED** [review]

File: tests/memory_tests/src/memory_tests_helper/memory_counter.cpp [hunk 4]
```diff
@@ -72,32 +72,34 @@ namespace MemoryTest {
...
-                res += c;
-        if (res.empty())
-            // If number wasn't found return -1
-            return -1;
-        return std::stoi(res);
-    }
-
-    size_t getSystemDataByName(char *name) {
-        FILE *file = fopen("/proc/self/status", "r");
+    size_t getSystemDataByName(char* name) {
```
Comment (ilyachur) [inline tests/memory_tests/src/memory_tests_helper/memory_counter.cpp:75]: In general memory tests could reuse implementation from the utils library

Comment (vurusovs) [reply tests/memory_tests/src/memory_tests_helper/memory_counter.cpp:75]: Yes, but historically it was appeared in stress tests first 😄

File: tests/memory_tests/src/memory_tests_helper/memory_counter.cpp [hunk 5]
```diff
@@ -27,8 +27,8 @@ namespace MemoryTest {
     static PROCESS_MEMORY_COUNTERS getMemoryInfo() {
         static PROCESS_MEMORY_COUNTERS pmc;
         pmc.cb = sizeof(PROCESS_MEMORY_COUNTERS);
-        GetProcessMemoryInfo(GetCurrentProcess(),&pmc, pmc.cb);
-        return pmc;
+        if (!GetProcessMemoryInfo(GetCurrentProcess(), &pmc, pmc.cb))
+            throw std::runtime_error("Can't get system memory values");
```
Comment (tiger100256-hu) [inline tests/memory_tests/src/memory_tests_helper/memory_counter.cpp:31]: "return pmc" is lost, will cause core dump on windows when run test

File: tests/stress_tests/common/utils.cpp [hunk 6]
```diff
@@ -33,25 +33,12 @@ std::string fileNameNoExt(const std::string &filepath) {
...
-}
-
 #ifdef _WIN32
 static PROCESS_MEMORY_COUNTERS getMemoryInfo() {
     static PROCESS_MEMORY_COUNTERS pmc;
     pmc.cb = sizeof(PROCESS_MEMORY_COUNTERS);
-    GetProcessMemoryInfo(GetCurrentProcess(),&pmc, pmc.cb);
-    return pmc;
+    if (!GetProcessMemoryInfo(GetCurrentProcess(), &pmc, pmc.cb))
+        throw std::runtime_error("Can't get system memory values");
```
Comment (tiger100256-hu) [inline tests/stress_tests/common/utils.cpp:41]: same issue

Comment (vurusovs) [reply tests/stress_tests/common/utils.cpp:41]: Good catch! Thanks, will be fixed

File: tests/memory_tests/src/memory_tests_helper/memory_counter.cpp (hunk 5 above)
Comment (vurusovs) [reply tests/memory_tests/src/memory_tests_helper/memory_counter.cpp:31]: Good catch! Thanks, will be fixed

**Pull request merged**